src/main/java/com/troubleshooting/troubleshootingtool/ContainerDataRetrieval.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Retrieves the data returned from the Docker Client and packs the data in the appropriate data models
+ */
+package com.troubleshooting.troubleshootingtool;
+
+import com.troubleshooting.model.ConjurLogModel;
+import com.troubleshooting.model.EnvironmentModel;
+import com.troubleshooting.model.EnvironmentsModel;
+import com.troubleshooting.model.LogsModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public final class ContainerDataRetrieval implements IDataRetrieval {
+    private ContainerDataAccess access;
+    private ConjurLogModel log;
+    private LogsModel logs;
+    private EnvironmentModel env;
+    private EnvironmentsModel envs;
+
+    static String STDERR = "STDERR";
+
+    @Autowired
+    public ContainerDataRetrieval(ContainerDataAccess access, ConjurLogModel log, LogsModel logs, EnvironmentModel env, EnvironmentsModel envs) {
+        this.access = access;
+        this.log = log;
+        this.logs = logs;
+        this.env = env;
+        this.envs = envs;
+    }
+
+    @Override
+    public LogsModel getLogs(String containerID, String query) {
+        List<String> loggingFrames = new ArrayList<String>();
+
+        try {
+            loggingFrames = access.getLogs(containerID, query);
+
+            loggingFrames.stream().forEach(entry -> extractImportantPartsOfLogEntry(entry));
+        } catch (NullPointerException e) {
+            System.out.println(e);
+        }
+        return logs;
+    }
+
+    private void extractImportantPartsOfLogEntry(String logInstance) {
+        String[] logEntry = logInstance.split(" ", 5);
+        if (!logEntry[0].contains(STDERR) && !logEntry[0].isEmpty() && logEntry.length >= 5) {
+            if (logEntry[1].contains("origin") && logEntry[2].contains("request_id") && logEntry[3].contains("tid")) {
+                log.setOrigin(logEntry[1]);
+                log.setRequest_id(logEntry[2]);
+                log.setThread_id(logEntry[3]);
+                log.setMessage(logEntry[logEntry.length - 1]);
+                logs.add(log);
```

Comment:
add a method that accepts a collection in logs

Reply:
Why a collection of logs?
I changed the code to be:
```
        if (!logEntry[0].contains(STDERR) && !logEntry[0].isEmpty() && logEntry.length >= 5) {
            if (logEntry[1].contains("origin") && logEntry[2].contains("request_id") && logEntry[3].contains("tid")) {
                logCollection.add(new ConjurLogModel(logEntry[1], logEntry[2], logEntry[3], logEntry[logEntry.length - 1]));
            }
        }
```
So I add each log entry incrementally, not a collection all at once.

Reply:
is there other place in the code in which you add a single log model?